File: Algorithms/3. Longest Substring Without Repeating Characters/php/solution.php
<?php
class Solution
{

    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s)
    {
        $ans = 0;
        $mp = [];

        $i = 0;
        for ($j = 0; $j < strlen($s); $j++) {
            if (isset($mp[$s[$j]])) {
                $i = max($i, $mp[$s[$j]]);
            }
            $ans = max($ans, $j - $i + 1);
            $mp[$s[$j]] = $j + 1;
        }
        return $ans;
    }
}

File: Algorithms/3. Longest Substring Without Repeating Characters/golang/solution.go
func lengthOfLongestSubstring(s string) int {
	var ans int
	var mp = make(map[rune]int)

	var i int
	for j, c := range s {
		if val := mp[c]; val > 0 {
			i = max(val, i)
		}
		ans = max(ans, j-i+1)
		mp[c] = j + 1
	}
	return ans
}

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
